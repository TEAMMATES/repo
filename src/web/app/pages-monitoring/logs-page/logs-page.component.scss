.bg-form {
  background-color: #EAEFF5;
}

.no-border {
  border: 0;
}

.center {
  text-align: center;
}

.no-data-message {
  padding: 10px;
  text-align: center;
}

.inline {
  width: 200px;
  display: inline;
}

@media (max-width: 768px) {
  .col-6 {
    width: auto;
  }

  .col-4 {
    width: auto;
  }

  .col-1 {
    width: auto;
  }
}
