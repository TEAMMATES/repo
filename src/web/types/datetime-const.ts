export enum HoursConst {
    ZERO_HOURS = 0,
    TWELVE_HOURS = 12,
    ONE_DAY_HOURS = 24,
    THREE_DAYS_HOURS = 72,
    ONE_WEEK_HOURS = 168,
}

export enum SecondsConst {
    ONE_MINUTE_SECONDS = 60,
}

export enum MinutesConst {
    ONE_HOURS_MINUTES = 60,
}

export enum MsConst {
    ONE_SECOND_MILLISECONDS = 1000,
    TEN_SECOND_MILLISECONDS = 10000,
    ONE_MINUTE_MILLISECONDS = 60000,
    TEN_MINUTE_MILLISECONDS = 600000,
    FIFTEEN_MINUTE_MILLISECONDS = 900000,
    ONE_HOUR_MILLISECONDS = 3600000,
    ONE_WEEK_MILLISECONDS = 604800000,
}

export const getDefaultDateFormat = (): DateFormat => {
    return { year: 0, month: 0, day: 0 };
};

export const getLatestTimeFormat = (): TimeFormat => {
    return { hour: 23, minute: 59 };
};

export const getDefaultTimeFormat = (): TimeFormat => {
    return { hour: 0, minute: 0 };
};

/**
 * The date format.
 */
export interface DateFormat {
    year: number;
    month: number;
    day: number;
}

/**
 * The output format of the time picker.
 */
export interface TimeFormat {
    hour: number;
    minute: number;
}
