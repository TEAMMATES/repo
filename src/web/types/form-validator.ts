import { ApiConst, ApiStringConst } from './api-const';

/**
 * Represents the root FormValidator object of all form fields.
 */
export enum FormValidator {
  /**
   * Max length for the 'Course ID' field.
   */
  COURSE_ID_MAX_LENGTH = ApiConst.COURSE_ID_MAX_LENGTH,

  /**
   * Max length for the 'Course Name' field.
   */
  COURSE_NAME_MAX_LENGTH = ApiConst.COURSE_NAME_MAX_LENGTH,

  /**
   * Max length for the 'Student Name` field.
   */
  STUDENT_NAME_MAX_LENGTH = ApiConst.STUDENT_NAME_MAX_LENGTH,

  /**
   * Max length for the 'Section Name` field.
   */
  SECTION_NAME_MAX_LENGTH = ApiConst.SECTION_NAME_MAX_LENGTH,

  /**
   * Max length for the 'Team Name` field.
   */
  TEAM_NAME_MAX_LENGTH = ApiConst.TEAM_NAME_MAX_LENGTH,

  /**
   * Max length for the 'E-mail Address` field.
   */
  EMAIL_MAX_LENGTH = ApiConst.EMAIL_MAX_LENGTH,

  /**
   * Regex used to verify URLs as Angular does not have a built-in URL validator.
   */
  URL_REGEX = ApiStringConst.URL_REGEX,

  /**
   * Regex used to verify emails in the back-end.
   */
  EMAIL_REGEX = ApiStringConst.EMAIL_REGEX,
}
