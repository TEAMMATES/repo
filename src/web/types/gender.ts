/**
 * Represents the gender of a student.
 */
export enum Gender {
  MALE,
  FEMALE,
  OTHER,
}
export namespace Gender {

  export const values: () => string[] = (): string[] =>
      Object.keys(Gender).filter((key: string) => isNaN(key as any) && key !== 'values');

  export const enums: () => Gender[] = (): Gender[] => {
    const genders: Gender[] = [];
    genders.push(Gender.MALE);
    genders.push(Gender.FEMALE);
    genders.push(Gender.OTHER);
    return genders;
  };
}
