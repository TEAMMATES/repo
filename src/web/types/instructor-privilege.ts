import { InstructorPrivilege } from './api-output';

/**
 * The default instructor privilege.
 */
export const DEFAULT_INSTRUCTOR_PRIVILEGE: InstructorPrivilege = {
  canEditCourse: false,
  canDeleteCourse: false,
  canRestoreCourse: false,
  canModifySession: false,
  canModifyStudent: false,
  canSubmitSessionInSections: false,
  canModifyInstructor: false,
  canViewStudentInSections: false,
  canModifySessionCommentsInSections: false,
  canViewSessionInSections: false,
};
