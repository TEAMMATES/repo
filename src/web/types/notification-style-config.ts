import { NotificationStyle } from './api-request';

enum NotificationFontColor {
    WHITE = 'text-white',
    DARK = 'text-dark',
}

interface NotificationStyleConfig {
    font: NotificationFontColor,
    description: string,
}

/**
 * Maps between notification style and actual class/description to be used.
 *
 * Ref: https://getbootstrap.com/docs/4.5/utilities/colors/#background-color
 */
export const notificationStyleConfigMap: Record<NotificationStyle, NotificationStyleConfig> = {
    PRIMARY: { font: NotificationFontColor.WHITE, description: 'Primary (blue)' },
    SECONDARY: { font: NotificationFontColor.WHITE, description: 'Secondary (grey)' },
    SUCCESS: { font: NotificationFontColor.WHITE, description: 'Success (green)' },
    DANGER: { font: NotificationFontColor.WHITE, description: 'Danger (red)' },
    WARNING: { font: NotificationFontColor.DARK, description: 'Warning (yellow)' },
    INFO: { font: NotificationFontColor.WHITE, description: 'Info (cyan)' },
    LIGHT: { font: NotificationFontColor.DARK, description: 'Light' },
    DARK: { font: NotificationFontColor.WHITE, description: 'Dark' },
};
