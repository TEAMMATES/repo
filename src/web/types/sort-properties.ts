/**
 * Elements that tables are sorted by.
 */
export enum SortBy {
    /**
     * Nothing.
     */
    NONE,

    /**
     * Section Name.
     */
    SECTION_NAME,

    /**
     * Team name.
     */
    TEAM_NAME,

    /**
     * Student Name.
     */
    STUDENT_NAME,

    /**
     * The email of the student.
     */
    EMAIL,

    /**
     * The gender of the student.
     */
    STUDENT_GENDER,

    /**
     * Institution.
     */
    INSTITUTION,

    /**
     * Nationality.
     */
    NATIONALITY,

    /**
     * Join status
     */
     JOIN_STATUS,

    /**
     * Course ID.
     */
    COURSE_ID,

    /**
     * Course Name.
     */
    COURSE_NAME,

    /**
     * The creation time of the course.
     */
    COURSE_CREATION_DATE,

    /**
     * Completion status of feedback session.
     */
    SESSION_COMPLETION_STATUS,

    /**
     * Feedback session name.
     */
    SESSION_NAME,

    /**
     * Start time of the feedback session.
     */
    SESSION_START_DATE,

    /**
     * End time of the feedback session.
     */
    SESSION_END_DATE,

    /**
     * The creation time of the feedback session.
     */
    SESSION_CREATION_DATE,

    /**
     * The time when the feedback session is moved to recycle bin.
     */
    SESSION_DELETION_DATE,

    /**
     * Feedback question type.
     */
    QUESTION_TYPE,

    /**
     * Feedback question text (brief).
     */
    QUESTION_TEXT,

    /**
     * Team of the giver of the feedback response.
     */
    GIVER_TEAM,

    /**
     * Name of the giver of the feedback response.
     */
    GIVER_NAME,

    /**
     * Team of the recipient of the feedback response.
     */
    RECIPIENT_TEAM,

    /**
     * Name of the recipient of the feedback response.
     */
    RECIPIENT_NAME,
}

/**
 * Sort order.
 */
export enum SortOrder {
    /**
     * Descending sort order.
     */
    DESC,

    /**
     * Ascending sort order
     */
    ASC,
}
