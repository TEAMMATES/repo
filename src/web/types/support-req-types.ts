export interface SupportRequest extends SupportRequestRequest {
    trackingId: string,
    status: SupportReqStatus
}

export interface SupportRequestRequest {
    email: string,
    name: string,
    type: SupportReqEnquiryType,
    title: string,
    initial_msg: string,
    createdAt: number,
    updatedAt: number,
}

/**
 * The ordering in the supportRequestList is based on the order of this list
 */
export enum SupportReqStatus {
    NEW,
    AWAITING_ADMIN,
    AWAITING_USER,
    RESOLVED
}

/**
 * The ordering in the supportRequestList is based on the order of this list
 */
export enum SupportReqEnquiryType {
    NEW_ACCOUNT,
    GENERAL_HELP
}