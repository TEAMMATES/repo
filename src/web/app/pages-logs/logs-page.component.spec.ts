import { HttpClientTestingModule } from '@angular/common/http/testing';
import { async, ComponentFixture, TestBed } from '@angular/core/testing';
import { NgbModule } from '@ng-bootstrap/ng-bootstrap';
import { of } from 'rxjs';
import { LogService } from '../../services/log.service';
import { TimezoneService } from '../../services/timezone.service';
import { GeneralLogEntry } from '../../types/api-output';
import { LogsPageComponent } from './logs-page.component';
import { LogsPageModule } from './logs-page.module';
import Spy = jasmine.Spy;

describe('LogsPageComponent', () => {
  let component: LogsPageComponent;
  let fixture: ComponentFixture<LogsPageComponent>;
  let logService: LogService;
  let timezoneService: TimezoneService;

  beforeEach(async(() => {
    TestBed.configureTestingModule({
      imports: [NgbModule, LogsPageModule, HttpClientTestingModule],
    })
    .compileComponents();
  }));

  beforeEach(() => {
    fixture = TestBed.createComponent(LogsPageComponent);
    logService = TestBed.inject(LogService);
    timezoneService = TestBed.inject(TimezoneService);
    component = fixture.componentInstance;
    fixture.detectChanges();
  });

  it('should create', () => {
    expect(component).toBeTruthy();
  });

  it('should snap with default fields', () => {
    expect(fixture).toMatchSnapshot();
  });

  it('should snap when page is still loading', () => {
    component.isLoading = true;
    fixture.detectChanges();
    expect(fixture).toMatchSnapshot();
  });

  it('should snap when searching for details in search form', () => {
    const logSeverity: Set<string> = new Set();
    logSeverity.add('ERROR');
    component.formModel = {
      logsSeverity: logSeverity,
      logsDateFrom: { year: 2021, month: 6, day: 1 },
      logsTimeFrom: { hour: 23, minute: 59 },
      logsDateTo: { year: 2021, month: 6, day: 2 },
      logsTimeTo: { hour: 23, minute: 59 },
    };
    component.isLoading = false;
    component.isSearching = true;
    fixture.detectChanges();

    expect(fixture).toMatchSnapshot();
  });

  it('should search for logs when search button is clicked', () => {
    const testLog: GeneralLogEntry = {
      logName: 'stdout',
      severity: 'INFO',
      trace: 'testTrace',
      sourceLocation: {
        file: 'file',
        line: 10,
        function: 'function',
      },
      timestamp: 1549095330000,
      message: 'message',
    };
    const logSpy: Spy = spyOn(logService, 'searchLogs').and
        .returnValue(of({ logEntries: [testLog], nextPageToken: 'testToken' }));
    const timeSpy: Spy = spyOn(timezoneService, 'getResolvedTimestamp').and
        .returnValue(of({ timestamp: 0, message: '' }));

    component.isLoading = false;
    component.isSearching = false;
    const logSeverity: Set<string> = new Set();
    logSeverity.add('INFO');
    component.formModel = {
      logsSeverity: logSeverity,
      logsDateFrom: { year: 2021, month: 6, day: 1 },
      logsTimeFrom: { hour: 23, minute: 59 },
      logsDateTo: { year: 2021, month: 6, day: 2 },
      logsTimeTo: { hour: 23, minute: 59 },
    };
    fixture.detectChanges();

    fixture.debugElement.nativeElement.querySelector('#query-button').click();

    expect(timeSpy).toHaveBeenCalledTimes(2);
    expect(logSpy).toHaveBeenCalledWith({
      searchFrom: '0',
      searchUntil: '0',
      severities: 'INFO',
    });

    expect(component.searchResults.length).toEqual(1);
  });

  it('should disable load button if there is no next page token', () => {
    component.nextPageToken = '';
    component.isSearching = false;
    component.hasResult = true;
    fixture.detectChanges();
    const button: any = fixture.debugElement.nativeElement.querySelector('#load-button');
    expect(button.disabled).toBeTruthy();
  });
});
