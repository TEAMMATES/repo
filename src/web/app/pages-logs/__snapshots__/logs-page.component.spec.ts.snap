// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LogsPageComponent should snap when page is still loading 1`] = `
<tm-logs-page
  LOGS_RETENTION_PERIOD_IN_DAYS={[Function Number]}
  LOGS_RETENTION_PERIOD_IN_MILLISECONDS={[Function Number]}
  SEVERITIES={[Function Array]}
  dateToday={[Function Object]}
  earliestSearchDate={[Function Object]}
  formModel={[Function Object]}
  hasResult="false"
  isLoading={[Function Boolean]}
  isSearching="false"
  logService={[Function LogService]}
  nextPageToken=""
  previousQueryParams={[Function Object]}
  searchResults={[Function Array]}
  statusMessageService={[Function StatusMessageService]}
  timezoneService={[Function TimezoneService]}
>
  <h1>
    Logs
  </h1><p>
    This page allows you to view the logs in the past 30 days.
  </p><hr /><tm-loading-spinner>
    <div
      class="loading-container"
    >
      <div
        class="row spinner-border text-primary"
        role="status"
      />
      <div
        class="row text-secondary"
      >
        Loading...
      </div>
    </div>
  </tm-loading-spinner><div>
    
    
    <div
      class="center"
    >
      
    </div>
  </div>
</tm-logs-page>
`;

exports[`LogsPageComponent should snap when searching for details in search form 1`] = `
<tm-logs-page
  LOGS_RETENTION_PERIOD_IN_DAYS={[Function Number]}
  LOGS_RETENTION_PERIOD_IN_MILLISECONDS={[Function Number]}
  SEVERITIES={[Function Array]}
  dateToday={[Function Object]}
  earliestSearchDate={[Function Object]}
  formModel={[Function Object]}
  hasResult="false"
  isLoading="false"
  isSearching={[Function Boolean]}
  logService={[Function LogService]}
  nextPageToken=""
  previousQueryParams={[Function Object]}
  searchResults={[Function Array]}
  statusMessageService={[Function StatusMessageService]}
  timezoneService={[Function TimezoneService]}
>
  <h1>
    Logs
  </h1><p>
    This page allows you to view the logs in the past 30 days.
  </p><hr /><div
    class="card bg-form no-border"
  >
    <div
      class="card-body"
    >
      <div
        class="row form-group"
      >
        <div
          class="col-6"
        >
          <div
            class="col-12 mb-1"
          >
            <label
              class="font-weight-bold"
              for="logs-from-datepicker"
            >
              Search period from
            </label>
          </div>
          <div
            class="input-group"
          >
            <div
              class="col-md-6 input-group"
            >
              <input
                class="form-control ng-untouched ng-pristine ng-valid"
                id="logs-from-datepicker"
                ngbdatepicker=""
                type="text"
              />
              <button
                class="btn btn-light"
                type="button"
              >
                <i
                  class="fas fa-calendar-alt"
                />
              </button>
            </div>
            <div
              class="col-md-3"
              id="logs-from-timepicker"
            >
              <tm-time-picker>
                <select
                  class="form-control ng-untouched ng-pristine ng-valid"
                >
                  
                  <option
                    value="1: Object"
                  >
                     01:00H 
                  </option>
                  <option
                    value="2: Object"
                  >
                     02:00H 
                  </option>
                  <option
                    value="3: Object"
                  >
                     03:00H 
                  </option>
                  <option
                    value="4: Object"
                  >
                     04:00H 
                  </option>
                  <option
                    value="5: Object"
                  >
                     05:00H 
                  </option>
                  <option
                    value="6: Object"
                  >
                     06:00H 
                  </option>
                  <option
                    value="7: Object"
                  >
                     07:00H 
                  </option>
                  <option
                    value="8: Object"
                  >
                     08:00H 
                  </option>
                  <option
                    value="9: Object"
                  >
                     09:00H 
                  </option>
                  <option
                    value="10: Object"
                  >
                     10:00H 
                  </option>
                  <option
                    value="11: Object"
                  >
                     11:00H 
                  </option>
                  <option
                    value="12: Object"
                  >
                     12:00H 
                  </option>
                  <option
                    value="13: Object"
                  >
                     13:00H 
                  </option>
                  <option
                    value="14: Object"
                  >
                     14:00H 
                  </option>
                  <option
                    value="15: Object"
                  >
                     15:00H 
                  </option>
                  <option
                    value="16: Object"
                  >
                     16:00H 
                  </option>
                  <option
                    value="17: Object"
                  >
                     17:00H 
                  </option>
                  <option
                    value="18: Object"
                  >
                     18:00H 
                  </option>
                  <option
                    value="19: Object"
                  >
                     19:00H 
                  </option>
                  <option
                    value="20: Object"
                  >
                     20:00H 
                  </option>
                  <option
                    value="21: Object"
                  >
                     21:00H 
                  </option>
                  <option
                    value="22: Object"
                  >
                     22:00H 
                  </option>
                  <option
                    value="23: Object"
                  >
                     23:00H 
                  </option>
                  <option
                    value="0: Object"
                  >
                    23:59H
                  </option>
                  
                </select>
              </tm-time-picker>
            </div>
          </div>
        </div>
        <div
          class="col-4"
        >
          <div
            class="col-12"
          >
            <label
              class="font-weight-bold"
              for="log-severity-checkbox"
            >
              Log severity level
            </label>
          </div>
          
          <div
            class="form-check"
          >
            <label
              class="form-check-label"
              id="log-severity-checkbox"
            >
              <input
                class="form-check-input"
                type="checkbox"
              />
              <span>
                INFO
              </span>
            </label>
          </div>
          <div
            class="form-check"
          >
            <label
              class="form-check-label"
              id="log-severity-checkbox"
            >
              <input
                class="form-check-input"
                type="checkbox"
              />
              <span>
                WARNING
              </span>
            </label>
          </div>
          <div
            class="form-check"
          >
            <label
              class="form-check-label"
              id="log-severity-checkbox"
            >
              <input
                class="form-check-input"
                type="checkbox"
              />
              <span>
                ERROR
              </span>
            </label>
          </div>
        </div>
        <div
          class="col-1 mt-4"
        >
          <button
            class="float-right btn btn-primary"
            id="query-button"
          >
            Query
          </button>
        </div>
      </div>
      <div
        class="row form-group"
      >
        <div
          class="col-6"
        >
          <div
            class="col-12 mb-1"
          >
            <label
              class="font-weight-bold"
              for="logs-to-datepicker"
            >
              Search period until
            </label>
          </div>
          <div
            class="input-group"
          >
            <div
              class="col-md-6 input-group"
            >
              <input
                class="form-control ng-untouched ng-pristine ng-valid"
                id="logs-to-datepicker"
                ngbdatepicker=""
                type="text"
              />
              <button
                class="btn btn-light"
                type="button"
              >
                <i
                  class="fas fa-calendar-alt"
                />
              </button>
            </div>
            <div
              class="col-md-3"
              id="logs-to-timepicker"
            >
              <tm-time-picker>
                <select
                  class="form-control ng-untouched ng-pristine ng-valid"
                >
                  
                  <option
                    value="1: Object"
                  >
                     01:00H 
                  </option>
                  <option
                    value="2: Object"
                  >
                     02:00H 
                  </option>
                  <option
                    value="3: Object"
                  >
                     03:00H 
                  </option>
                  <option
                    value="4: Object"
                  >
                     04:00H 
                  </option>
                  <option
                    value="5: Object"
                  >
                     05:00H 
                  </option>
                  <option
                    value="6: Object"
                  >
                     06:00H 
                  </option>
                  <option
                    value="7: Object"
                  >
                     07:00H 
                  </option>
                  <option
                    value="8: Object"
                  >
                     08:00H 
                  </option>
                  <option
                    value="9: Object"
                  >
                     09:00H 
                  </option>
                  <option
                    value="10: Object"
                  >
                     10:00H 
                  </option>
                  <option
                    value="11: Object"
                  >
                     11:00H 
                  </option>
                  <option
                    value="12: Object"
                  >
                     12:00H 
                  </option>
                  <option
                    value="13: Object"
                  >
                     13:00H 
                  </option>
                  <option
                    value="14: Object"
                  >
                     14:00H 
                  </option>
                  <option
                    value="15: Object"
                  >
                     15:00H 
                  </option>
                  <option
                    value="16: Object"
                  >
                     16:00H 
                  </option>
                  <option
                    value="17: Object"
                  >
                     17:00H 
                  </option>
                  <option
                    value="18: Object"
                  >
                     18:00H 
                  </option>
                  <option
                    value="19: Object"
                  >
                     19:00H 
                  </option>
                  <option
                    value="20: Object"
                  >
                     20:00H 
                  </option>
                  <option
                    value="21: Object"
                  >
                     21:00H 
                  </option>
                  <option
                    value="22: Object"
                  >
                     22:00H 
                  </option>
                  <option
                    value="23: Object"
                  >
                     23:00H 
                  </option>
                  <option
                    value="0: Object"
                  >
                    23:59H
                  </option>
                  
                </select>
              </tm-time-picker>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div><div>
    
    
    <tm-loading-spinner>
      <div
        class="loading-container"
      >
        <div
          class="row spinner-border text-primary"
          role="status"
        />
        <div
          class="row text-secondary"
        >
          Loading...
        </div>
      </div>
    </tm-loading-spinner>
  </div>
</tm-logs-page>
`;

exports[`LogsPageComponent should snap with default fields 1`] = `
<tm-logs-page
  LOGS_RETENTION_PERIOD_IN_DAYS={[Function Number]}
  LOGS_RETENTION_PERIOD_IN_MILLISECONDS={[Function Number]}
  SEVERITIES={[Function Array]}
  dateToday={[Function Object]}
  earliestSearchDate={[Function Object]}
  formModel={[Function Object]}
  hasResult="false"
  isLoading="false"
  isSearching="false"
  logService={[Function LogService]}
  nextPageToken=""
  previousQueryParams={[Function Object]}
  searchResults={[Function Array]}
  statusMessageService={[Function StatusMessageService]}
  timezoneService={[Function TimezoneService]}
>
  <h1>
    Logs
  </h1><p>
    This page allows you to view the logs in the past 30 days.
  </p><hr /><div
    class="card bg-form no-border"
  >
    <div
      class="card-body"
    >
      <div
        class="row form-group"
      >
        <div
          class="col-6"
        >
          <div
            class="col-12 mb-1"
          >
            <label
              class="font-weight-bold"
              for="logs-from-datepicker"
            >
              Search period from
            </label>
          </div>
          <div
            class="input-group"
          >
            <div
              class="col-md-6 input-group"
            >
              <input
                class="form-control ng-untouched ng-pristine ng-valid"
                id="logs-from-datepicker"
                ngbdatepicker=""
                type="text"
              />
              <button
                class="btn btn-light"
                type="button"
              >
                <i
                  class="fas fa-calendar-alt"
                />
              </button>
            </div>
            <div
              class="col-md-3"
              id="logs-from-timepicker"
            >
              <tm-time-picker>
                <select
                  class="form-control ng-untouched ng-pristine ng-valid"
                >
                  
                  <option
                    value="1: Object"
                  >
                     01:00H 
                  </option>
                  <option
                    value="2: Object"
                  >
                     02:00H 
                  </option>
                  <option
                    value="3: Object"
                  >
                     03:00H 
                  </option>
                  <option
                    value="4: Object"
                  >
                     04:00H 
                  </option>
                  <option
                    value="5: Object"
                  >
                     05:00H 
                  </option>
                  <option
                    value="6: Object"
                  >
                     06:00H 
                  </option>
                  <option
                    value="7: Object"
                  >
                     07:00H 
                  </option>
                  <option
                    value="8: Object"
                  >
                     08:00H 
                  </option>
                  <option
                    value="9: Object"
                  >
                     09:00H 
                  </option>
                  <option
                    value="10: Object"
                  >
                     10:00H 
                  </option>
                  <option
                    value="11: Object"
                  >
                     11:00H 
                  </option>
                  <option
                    value="12: Object"
                  >
                     12:00H 
                  </option>
                  <option
                    value="13: Object"
                  >
                     13:00H 
                  </option>
                  <option
                    value="14: Object"
                  >
                     14:00H 
                  </option>
                  <option
                    value="15: Object"
                  >
                     15:00H 
                  </option>
                  <option
                    value="16: Object"
                  >
                     16:00H 
                  </option>
                  <option
                    value="17: Object"
                  >
                     17:00H 
                  </option>
                  <option
                    value="18: Object"
                  >
                     18:00H 
                  </option>
                  <option
                    value="19: Object"
                  >
                     19:00H 
                  </option>
                  <option
                    value="20: Object"
                  >
                     20:00H 
                  </option>
                  <option
                    value="21: Object"
                  >
                     21:00H 
                  </option>
                  <option
                    value="22: Object"
                  >
                     22:00H 
                  </option>
                  <option
                    value="23: Object"
                  >
                     23:00H 
                  </option>
                  <option
                    value="0: Object"
                  >
                    23:59H
                  </option>
                  
                </select>
              </tm-time-picker>
            </div>
          </div>
        </div>
        <div
          class="col-4"
        >
          <div
            class="col-12"
          >
            <label
              class="font-weight-bold"
              for="log-severity-checkbox"
            >
              Log severity level
            </label>
          </div>
          
          <div
            class="form-check"
          >
            <label
              class="form-check-label"
              id="log-severity-checkbox"
            >
              <input
                class="form-check-input"
                type="checkbox"
              />
              <span>
                INFO
              </span>
            </label>
          </div>
          <div
            class="form-check"
          >
            <label
              class="form-check-label"
              id="log-severity-checkbox"
            >
              <input
                class="form-check-input"
                type="checkbox"
              />
              <span>
                WARNING
              </span>
            </label>
          </div>
          <div
            class="form-check"
          >
            <label
              class="form-check-label"
              id="log-severity-checkbox"
            >
              <input
                class="form-check-input"
                type="checkbox"
              />
              <span>
                ERROR
              </span>
            </label>
          </div>
        </div>
        <div
          class="col-1 mt-4"
        >
          <button
            class="float-right btn btn-primary"
            id="query-button"
          >
            Query
          </button>
        </div>
      </div>
      <div
        class="row form-group"
      >
        <div
          class="col-6"
        >
          <div
            class="col-12 mb-1"
          >
            <label
              class="font-weight-bold"
              for="logs-to-datepicker"
            >
              Search period until
            </label>
          </div>
          <div
            class="input-group"
          >
            <div
              class="col-md-6 input-group"
            >
              <input
                class="form-control ng-untouched ng-pristine ng-valid"
                id="logs-to-datepicker"
                ngbdatepicker=""
                type="text"
              />
              <button
                class="btn btn-light"
                type="button"
              >
                <i
                  class="fas fa-calendar-alt"
                />
              </button>
            </div>
            <div
              class="col-md-3"
              id="logs-to-timepicker"
            >
              <tm-time-picker>
                <select
                  class="form-control ng-untouched ng-pristine ng-valid"
                >
                  
                  <option
                    value="1: Object"
                  >
                     01:00H 
                  </option>
                  <option
                    value="2: Object"
                  >
                     02:00H 
                  </option>
                  <option
                    value="3: Object"
                  >
                     03:00H 
                  </option>
                  <option
                    value="4: Object"
                  >
                     04:00H 
                  </option>
                  <option
                    value="5: Object"
                  >
                     05:00H 
                  </option>
                  <option
                    value="6: Object"
                  >
                     06:00H 
                  </option>
                  <option
                    value="7: Object"
                  >
                     07:00H 
                  </option>
                  <option
                    value="8: Object"
                  >
                     08:00H 
                  </option>
                  <option
                    value="9: Object"
                  >
                     09:00H 
                  </option>
                  <option
                    value="10: Object"
                  >
                     10:00H 
                  </option>
                  <option
                    value="11: Object"
                  >
                     11:00H 
                  </option>
                  <option
                    value="12: Object"
                  >
                     12:00H 
                  </option>
                  <option
                    value="13: Object"
                  >
                     13:00H 
                  </option>
                  <option
                    value="14: Object"
                  >
                     14:00H 
                  </option>
                  <option
                    value="15: Object"
                  >
                     15:00H 
                  </option>
                  <option
                    value="16: Object"
                  >
                     16:00H 
                  </option>
                  <option
                    value="17: Object"
                  >
                     17:00H 
                  </option>
                  <option
                    value="18: Object"
                  >
                     18:00H 
                  </option>
                  <option
                    value="19: Object"
                  >
                     19:00H 
                  </option>
                  <option
                    value="20: Object"
                  >
                     20:00H 
                  </option>
                  <option
                    value="21: Object"
                  >
                     21:00H 
                  </option>
                  <option
                    value="22: Object"
                  >
                     22:00H 
                  </option>
                  <option
                    value="23: Object"
                  >
                     23:00H 
                  </option>
                  <option
                    value="0: Object"
                  >
                    23:59H
                  </option>
                  
                </select>
              </tm-time-picker>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div><div>
    
    
    <div
      class="center"
    >
      
    </div>
  </div>
</tm-logs-page>
`;
